compiler/rustc_codegen_llvm/src/llvm/ffi.rs

```diff
@@ -2417,3 +2418,13 @@ unsafe extern "C" {
 
     pub fn LLVMRustIsECObject(buf_ptr: *const u8, buf_len: usize) -> bool;
 }
+
+impl From<SymbolVisibility> for Visibility {
+    fn from(value: SymbolVisibility) -> Self {
+        match value {
+            SymbolVisibility::Hidden => Visibility::Hidden,
+            SymbolVisibility::Protected => Visibility::Protected,
+            SymbolVisibility::Interposable => Visibility::Default,
+        }
+    }
+}
```

Comment:
Could you move this `From` impl just after the definition of `Visibility` at the top of this file; and instead of a `From` impl just have a `from_generic` associated function like it's done in the rest of the file.